src/components/browse.js

```diff
@@ -0,0 +1,25 @@
+import React, { Component } from 'react';
+import { render } from 'react-dom';
+
+
+class Browse extends Component {
+    render(){
```

Comment:
more a design question: an object-oriented principle of React is to let Component carry its data via props and states. I don't see we are doing that (correct me if i am wrong). 

For example, the current active topic to browse (depending on whether button user just pressed) should be state for Browse class, and then when that state is changed, we trigger the GitHub search API, get the results, pass the results as props to child component which represents the browse result box, and let it re-render?

Reply:
Your suggestion would definitely be the more 'react' way of implementing this, however it would require me to re-implement the widget code that we are using. For the sake of conserving time, I have gotten it working through a hybrid implementation of react and the widget (since the main value prop of using react was increased testability). I am open for a discussion on prioritizing the remaining work and figuring out how much additional time should be spent on re-implementing these the react way